File: C#/Lab4/TTime.cs
using System;

namespace lab4
{
    public class TTime : TTriad
    {
        public TTime(int n1, int n2, int n3) : base(n1, n2, n3)
        {
        }

        public override void IncreaseNumber1(int n)
        {
            Number1 += n;
            while (Number1 >= 24)
            {
                Number1 -= 24;
            }
        }

        public override void DecreaseNumber1(int n)
        {
            Number1 -= n;
            while (Number1 < 0)
            {
                Number1 += 24;
            }
        }

        public override void IncreaseNumber2(int n)
        {
            Number2 += n;
            while (Number2 >= 60)
            {
                Number2 -= 60;
                IncreaseNumber1(1);
            }
        }

        public override void DecreaseNumber2(int n)
        {
            Number2 -= n;
            while (Number2 < 0)
            {
                Number2 += 60;   
                DecreaseNumber1(1);
            }
        }

        public override void IncreaseNumber3(int n)
        {
            Number3 += n;
            while (Number3 >= 60)
            {
                Number3 -= 60;
                IncreaseNumber2(1);
            }
        }

        public override void DecreaseNumber3(int n)
        {
            Number3 -= n;
            while (Number3 < 0)
            {
                Number3 += 60;   
                DecreaseNumber2(1);
            }
        }

        public void Output()
        {
            Console.WriteLine($"{Funcs.ChangeFormat(Number1)}:{Funcs.ChangeFormat(Number2)}:{Funcs.ChangeFormat(Number3)}");
        }
    }
}

File: Python/Lab4/TTime.py
from TTriad import TTriad
import Funcs


class TTime(TTriad):
    def __init__(self, number1, number2, number3):
        super().__init__(number1, number2, number3)

    def IncreaseNumber1(self, n):
        self.Number1 += n
        while (self.Number1 >= 24):
            self.Number1 -= 24

    def DecreaseNumber1(self, n):
        self.Number1 -= n
        while self.Number1 < 0:
            self.Number1 += 24

    def IncreaseNumber2(self, n):
        self.Number2 += n
        while self.Number2 >= 60:
            self.Number2 -= 60
            self.IncreaseNumber1(1)

    def DecreaseNumber2(self, n):
        self.Number2 -= n
        while self.Number2 < 0:
            self.Number2 += 60
            self.DecreaseNumber1(1)

    def IncreaseNumber3(self, n):
        self.Number3 += n
        while self.Number3 >= 60:
            self.Number3 -= 60
            self.IncreaseNumber2(1)

    def DecreaseNumber3(self, n):
        self.Number3 -= n
        while self.Number3 < 0:
            self.Number3 += 60
            self.DecreaseNumber2(1)


    def Output(self):
        print(
            f"{Funcs.ChangeFormat(self.Number1)}:{Funcs.ChangeFormat(self.Number2)}:{Funcs.ChangeFormat(self.Number3)}")